### ISSUE TITLE
How to convert the similarity_scores from tensor array to a score?

### ISSUE THREAD

**Issue by LionLion888** [mk-minchul/AdaFace#121]
Hi,

I try to simply go through the AdaFace project, and get the similarity score as follows,

similarity_scores:
tensor([[1.0000, 0.5508],
        [0.5508, 1.0000]], grad_fn=<MmBackward0>)

But how could I convert the outcome to a score? such as follows,

similarity_scores:
0.985213

**Comment by ayush0x00** [comment]
For each image, you'll be getting a 1x512 feature vector. You can use `torch.squeeze()` to remove the first dimension and you can use `torch.nn.functionality.cosine_similarity()` to get the corresponding score. Other methods such as averaging the dot product of individual components, min-max, etc can also be used. The cosine similarity can be negative also, so for most of the cases, normalizing using min-max normalization will do the job.
